framework/src/modules/chain/blocks/blocks.js

```diff
@@ -452,23 +464,27 @@ class Blocks extends EventEmitter {
 
 	async _receiveBlockFromNetworkV2(block) {
 		// Current time since Lisk Epoch
-		// Better to do it here rather than in the Sequence so receiving time is more accurate
-		const newBlockReceivedAt = this.slots.getTime();
+		block.receivedAt = this.slots.getSlotTime(this.slots.getSlotNumber());
 
 		// Execute in sequence
 		return this.sequence.add(async () => {
 			this._shouldNotBeActive();
 			this._isActive = true;
-			const result = await this._forkChoiceTask(block, newBlockReceivedAt);
-			this._isActive = false;
-			return result;
+			try {
+				await this._forkChoiceTask(block);
```

Comment:
this was throwing before, but it should be caught?

Reply:
I wonder what happens when an error is thrown in a Sequence? Do you have more info about the expected behavior?

Reply:
If error happens within the sequence, outside promise will return `Promise.reject(err)`.
I doubled check the V1 implementation, and it's correct to catch the error 🙏